### DISCUSSION TITLE
TCustomButton and OnMouseEvent

### DISCUSSION THREAD

**Discussion by fredvs** [mse-org/mseide-msegui#80] [General]
Do you agree to make **onmouseevent()** published for **tcustombutton**?

I would like to be able to control things when something is done with the mouse on a Tbutton (not only execute).
But property **onmouseevent()**  is not published so no access to mouse-event.
I did try publishing it and it works like charm.

By the way, I was not able to make work **tbutton.onfocus, tbutton.ondefocus, onbeforeexecute.**

**Comment by rchastain** [comment]
If you look at the **hoverframe** example (I don't remember where it comes from, probably from MSEUniverse), you will see that the button reacts to mouse move. So I am not sure that a modification is necessary for what you want to do.

**Comment by fredvs** [comment]
OK, I will see, thanks.

**Comment by fredvs** [comment]
https://github.com/mse-org/mseide-msegui/discussions/68#discussioncomment-8712546

**Comment by rchastain** [comment]
Bravo Fred!

No idea how it works, but it works. :)

By the way, how do you define BGRABITMAPDIR? To build your project, I had to define it in project options, macros. What is your method?

**Comment by fredvs** [comment]
>By the way, how do you define BGRABITMAPDIR? To build your project, I had to define it in project options, macros. What is your method?

With MSEide, in "Configure MSEide" tab, create a new "Global Macro" (bottom of form) with name:  BGRABITMAPDIR and value: the directory of BGRABitmap.

![settings_mseide](https://github.com/mse-org/mseide-msegui/assets/3421249/dfe7abc1-7df5-4cb4-8089-db7e2fbc1836)

With ideU, in "General Settings" path tab, fill the directory of BGRABitmap.

![settings_ideu](https://github.com/mse-org/mseide-msegui/assets/3421249/02b760bf-af8e-4e68-9fd0-1ea6e4713f20)
